.ci/ci.R

```diff
@@ -43,12 +43,27 @@ function (repos, type = getOption("pkgType"), ver)
   res
 }
 
+local.extract_dependency_package_names = function (x) {
+  ## do not filter out R like tools:::.extract_dependency_package_names, used for web/$pkg/index.html
+  if (is.na(x))
+    return(character())
+  x <- unlist(strsplit(x, ",[[:space:]]*"))
+  x <- sub("[[:space:]]*([[:alnum:].]+).*", "\\1", x)
+  x[nzchar(x)]
+}
+
+rev.dependencies = function(pkg, available_pkgs, which=c("Imports", "Depends")) {
+  dependencies = available_pkgs[pkg, which]
+  x = dependencies[!is.na(dependencies)]
```

Comment:
here can you please insert a comment explaining why tools::dependsOnPkgs is not sufficient?

Reply:
Maybe I misunderstood something but `tools:dependsOnPkgs(x)` shows the relation for results depends on x (and only does so for installed packages). However, we want the inverse with x depends/imports result.

Reply:
> only does so for installed packages

This part is easy to overcome: `tools::dependsOnPkgs("data.table", installed=available.packages())`.

Anyway, you're correct that's going in the wrong direction; `tools::package_dependencies()` traverses the graph in the right direction:

```r
tools::package_dependencies("data.table", recursive=TRUE, which="all") # overkill
# second-level instead
tools::package_dependencies("data.table", which="all") |>
  unlist() |> tools::package_dependencies() |> # could use which="all" here too?
  unlist(use.names=FALSE) |> setdiff(tools:::.get_standard_package_names()$base)
```